src/legacy/core_plugins/kibana/public/discover/embeddable/search_embeddable.ts

```diff
@@ -96,6 +160,9 @@ export class SearchEmbeddable extends Embeddable {
 
     const timeRangeSearchSource = this.searchScope.searchSource.create();
     timeRangeSearchSource.setField('filter', () => {
+      if (!this.searchScope || !this.timeRange) {
```

Comment:
Maybe a better way to not need all those checks, would be just first creating that scope on a local variable, in this function and then assign at the end to the class:

```ts
const searchScope: SearchScope = this.$rootScope.$new();
searchScope.description = this.savedSearch.description;
// ...
searchScope.setSortOrder = (columnName, direction) => {
  // no check now required
  searchScope.sort = this.customization.sort = [ columnName, direction ];
}
// ...
this.searchScope = searchScope;
```

Reply:
But since we're checking in a lot of places still for existence of `columns` not sure if it really adds too much to the readability. So feel free to treat this as you wish.